cctp_primitives/src/proving_system/mod.rs

```diff
@@ -441,4 +441,165 @@ pub fn check_matching_proving_system_type(
     let vk_ps_type = vk.get_proving_system_type();
 
     proof_ps_type == vk_ps_type
+}
+
+use marlin::ahp::indexer::IndexInfo;
+
+/// Checks that size of proof and vk for a circuit with given segment_size, indexer_info, proof_type and zk,
+/// are smaller than, respectively, max_proof_size and max_vk_size.
+pub fn check_proof_vk_size(
+    segment_size: usize,
+    info: IndexInfo<FieldElement>,
+    zk: bool,
+    proof_type: ProvingSystem,
+    max_proof_size: usize,
+    max_vk_size: usize,
+) -> bool 
+{
+    let (proof_size, vk_size) = compute_proof_vk_size(segment_size, info, zk, proof_type);
+    proof_size <= max_proof_size && vk_size <= max_vk_size
+}
+
+/// Compute size of proof and vk.
+/// TODO: Currently, zk = false, gives 33 bytes more and vk size is 7 bytes more. Fix it. 
+pub(crate) fn compute_proof_vk_size(
+    segment_size: usize,
+    info: IndexInfo<FieldElement>,
+    zk: bool,
+    proof_type: ProvingSystem,
+) -> (usize, usize) 
+{
+    // Compute config data
+    let zk_bound: usize = if zk { 1 } else { 0 };
+    let segment_size = segment_size.next_power_of_two();
+    let h = std::cmp::max(info.num_constraints.next_power_of_two(), info.num_variables.next_power_of_two());
+    let k = info.num_non_zero.next_power_of_two();
+
+    // Compute num segments
+    let w_z_a_b_segs = ((h + 2 * zk_bound) as f64/segment_size as f64).ceil() as usize;
```

Comment:
I double-checked the computation of the gauged w(X) polynomial in `` z(X) = x(X)+w(X)*vanishing_poly(input_domain)``. In fact, the prover sets w(X) to zero on the input domain, and then does the division by the vanishing polynomial of the input domain. Hence the formula does not hold if zk=0: here ``w_segs = (h - input_domain_size)/segment_size.ceil()``.

Reply:
if zk =0 and 1:
deg w(X)  = (h + 2*zk_bound - input_domain_size)/segment_size,
because even for zk=1 the Marlin prover divides by the vanishing polynomial of the input domain.

Reply:
Done